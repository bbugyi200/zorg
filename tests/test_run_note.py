"""Tests for the zorg project's 'note' CLI command."""

from __future__ import annotations

from pathlib import Path
from typing import Optional

from pytest import mark, param
from syrupy.assertion import SnapshotAssertion as Snapshot

from . import common as c


params = mark.parametrize


@params(
    "zid,src_name,dest_name,mutate",
    [
        param("240510#02", "basic.zo", "property.zo", None, id="basic"),
        param(
            "240510#0Q",
            "property.zo",
            "tags_and_ids.zo",
            "~",
            id="canceled_multiline",
        ),
        param(
            "301231#X1",
            "tags_and_ids.zo",
            "priority.zo",
            "x",
            id="done_inherited_tags",
        ),
    ],
)
def test_note_move(
    main: c.MainType,
    db_zettel_dir: Path,
    snapshot: Snapshot,
    zid: str,
    src_name: str,
    dest_name: str,
    mutate: Optional[str],
) -> None:
    """Tests the 'note move' command."""
    main_args = ["--dir", str(db_zettel_dir), "note", "move", zid, dest_name]
    if mutate is not None:
        main_args.append(mutate)
    exit_code = main(*main_args)

    src_path = db_zettel_dir / src_name
    dest_path = db_zettel_dir / dest_name
    assert exit_code == 0
    assert snapshot == src_path.read_text()
    assert snapshot == dest_path.read_text()
