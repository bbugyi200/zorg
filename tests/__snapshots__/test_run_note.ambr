# serializer version: 1
# name: test_note_move[YAMLConfigFile-basic]
  '''
  # Some test file.
  #
  # blah blah blah
  
  - 240510#00 First note in first block
  - 240510#01 Second note in first block
  
  ################################ Foobar Rocks!
  o 000103#00 Old todo
  o 000102#00 Older todo
  
  '''
# ---
# name: test_note_move[YAMLConfigFile-basic].1
  '''
  # A zorg file containing properties.
  
  - 240510#0P This note has a single-line bullet property.
    * foo:: bar baz buz
  - 240510#0Q q:: Can a note headline use a bullet property or do bullet
    properties need to be defined _under_ the headline?
    * a:: Yes! Bullet properties are powerful!
  ~ 240613 240510#0R A canceled todo with a multiline explanation
    * x:: This todo was not really canceled. It is an example todo used to
      demonstrate a multiline bullet property.
  - 240629#00 A note with a bullet property that contains a sublist
    * url::
      - https://www.foobar.com
      - https://www.bazbuz.com
  
  ################################ p::1
  - 240330#02 This note contains a key::value pair.
  o 240510#0K This todo is @due today! | due::2024-03-13
  - 240510#0M This [type:: awesome note] has an inline property.
  o 240510#0N This recurring todo also uses an inline property | [recur::M+Th]
  o P3 240510#02 bazbuz is ok too...
  
  '''
# ---
# name: test_note_move[YAMLConfigFile-canceled_multiline]
  '''
  # A zorg file containing properties.
  
  - 240510#0P This note has a single-line bullet property.
    * foo:: bar baz buz
  ~ 240613 240510#0R A canceled todo with a multiline explanation
    * x:: This todo was not really canceled. It is an example todo used to
      demonstrate a multiline bullet property.
  - 240629#00 A note with a bullet property that contains a sublist
    * url::
      - https://www.foobar.com
      - https://www.bazbuz.com
  
  ################################ p::1
  - 240330#02 This note contains a key::value pair.
  o 240510#0K This todo is @due today! | due::2024-03-13
  - 240510#0M This [type:: awesome note] has an inline property.
  o 240510#0N This recurring todo also uses an inline property | [recur::M+Th]
  o P3 240510#02 bazbuz is ok too...
  
  '''
# ---
# name: test_note_move[YAMLConfigFile-canceled_multiline].1
  '''
  # A file with lots of #tags and zorg IDs.
  
  ################################ +foobar @fat_pig
  - 240408#0X Some note
  - 240509 240408#0Y Some note with a modify date
  
  ======================== +bazbuz #thoughts
  - 301231#X0 When I talk to %pig I always want to @puke. | ID::pig_is_gross
  - 301231#X1 @READ Some article
    * RID::some_article
    * url::http://www.example.com
  
  ################################ +bad
  ======================== +to_the_bone
  - 240408#N2 Da nah nah nuh...
  ======================== +2_be_or_not_2_be +123
  - 240510#29 Projects can start with a number, but CANNOT be all numbers
  ~ 240510#0Q q:: Can a note headline use a bullet property or do bullet
    properties need to be defined _under_ the headline?
    * a:: Yes! Bullet properties are powerful!
  
  '''
# ---
# name: test_note_move[YAMLConfigFile-done_inherited_tags]
  '''
  # A file with lots of #tags and zorg IDs.
  
  ################################ +foobar @fat_pig
  - 240408#0X Some note
  - 240509 240408#0Y Some note with a modify date
  
  ======================== +bazbuz #thoughts
  - 301231#X0 When I talk to %pig I always want to @puke. | ID::pig_is_gross
  
  ################################ +bad
  ======================== +to_the_bone
  - 240408#N2 Da nah nah nuh...
  ======================== +2_be_or_not_2_be +123
  - 240510#29 Projects can start with a number, but CANNOT be all numbers
  ~ 240510#0Q q:: Can a note headline use a bullet property or do bullet
    properties need to be defined _under_ the headline?
    * a:: Yes! Bullet properties are powerful!
  
  '''
# ---
# name: test_note_move[YAMLConfigFile-done_inherited_tags].1
  '''
  # A zorg file with todos that have priorities and dates.
  
  o P0 240313#03 High Priority TODO
  o P1 240313#04 Medium Priority TODO
  o P2 240313#05 Low Priority TODO
  x 301231#X1 +bazbuz +foobar #tags #thoughts @fat_pig @READ Some article
    * RID::some_article
    * url::http://www.example.com
  
  '''
# ---
